src/solver/optimisation/constraints/RampingDecreaseRate.cpp

```diff
@@ -0,0 +1,48 @@
+#include "RampingDecreaseRate.h"
+
+void RampingDecreaseRate::add(int pays, int cluster, int clusterIndex, int pdt, bool Simulation)
+{
+    if (!Simulation)
+    {
+        const PALIERS_THERMIQUES& PaliersThermiquesDuPays = problemeHebdo->PaliersThermiquesDuPays[pays];
+        double maxDownwardPowerRampingRate = PaliersThermiquesDuPays.maxDownwardPowerRampingRate[clusterIndex];
+        double pmaxDUnGroupeDuPalierThermique = PaliersThermiquesDuPays.PmaxDUnGroupeDuPalierThermique[clusterIndex];
+        // constraint : P(t) - P(t-1) + R^- * M(t) + u * M^-(t) + u * M^--(t) > 0
+        if (pdt > 0)
+        {
+            builder.updateHourWithinWeek(pdt)
+              .DispatchableProduction(cluster, 1.0)
+              .DispatchableProduction(cluster, -1.0, -1, problemeHebdo->NombreDePasDeTempsPourUneOptimisation)
+              .NumberOfDispatchableUnits(cluster, maxDownwardPowerRampingRate)
+              .NumberStoppingDispatchableUnits(cluster, pmaxDUnGroupeDuPalierThermique)
+              .NumberBreakingDownDispatchableUnits(cluster, pmaxDUnGroupeDuPalierThermique)
+              .greaterThan();
+        }
```

Comment:
You shouldn't need this `if..else` since `DispatchableProduction` has arguments to take this into account.

Reply:
Indeed, commit [624f2ce](https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/1817/commits/624f2ce5436532ef019be957135c5cbb154c5a60) fixed this.